snuba/migrations/connect.py

```diff
@@ -55,3 +58,45 @@ def check_clickhouse(clickhouse: ClickhousePool) -> None:
         raise InvalidClickhouseVersion(
             f"Snuba requires minimum Clickhouse version {CLICKHOUSE_SERVER_MIN_VERSION} ({clickhouse.host}:{clickhouse.port} - {ver})"
         )
+
+
+def check_for_inactive_replicas() -> None:
+    """
+    Checks for inactive replicas and raise InactiveClickhouseReplica if any are found.
+    """
+
+    storage_keys = [
+        storage_key
+        for storage_key in sorted(
+            get_all_storage_keys(), key=lambda storage_key: storage_key.value
+        )
+        if get_storage(storage_key).get_storage_set_key() not in DEV_STORAGE_SETS
+        or ENABLE_DEV_FEATURES
+    ]
+
+    checked_nodes = set()
+
+    for storage_key in storage_keys:
+        storage = get_storage(storage_key)
+        cluster = storage.get_cluster()
+
+        query_node = cluster.get_query_node()
+        local_nodes = cluster.get_local_nodes()
+        distributed_nodes = cluster.get_distributed_nodes()
+
+        for node in (*local_nodes, *distributed_nodes, query_node):
+            if (node.host_name, node.port) in checked_nodes:
+                continue
+            checked_nodes.add((node.host_name, node.port))
+
+            conn = cluster.get_node_connection(ClickhouseClientSettings.MIGRATE, node)
+            tables_with_inactive = conn.execute(
+                f"SELECT table, total_replicas, active_replicas FROM system.replicas "
+                f"WHERE active_replicas < total_replicas AND database ='{conn.database}'",
+            ).results
+
+            for table, total_replicas, active_replicas in tables_with_inactive:
+                raise InactiveClickhouseReplica(
```

Comment:
One thing to consider is how a user might be able to help themselves when they see this problem. Should we point them to the clickhouse documentation on how to remove it?

Also, since you are checking for all tables, this will only tell them the first cluster to fix. What if there are 3 other clusters which need to be fixed? They would be in a loop where they fix one, try again, fix second, try again, fix third, try again and succeed. Can we raise an exception which lists all clusters which need to be fixed?

Reply:
I fixed it to list all inactive replicas. 

But I think it may be tricky to point to the Clickhouse documentation because there are many reasons why a replica may be inactive and there is not exactly one solution to fix them. It's better to take it case by case.